pkg/network/ebpf/c/protocols/https.h

```diff
@@ -152,31 +160,22 @@ static __always_inline void map_ssl_ctx_to_sock(struct sock *skp) {
  */
 static __always_inline tls_offsets_data_t* get_offsets_data() {
     struct task_struct *t = (struct task_struct *) bpf_get_current_task();
-    struct mm_struct *mm;
-    struct file *exe_file;
-    struct inode *inode;
-    struct super_block *sb;
+    go_tls_offsets_data_key_t key;
     dev_t dev_id;
 
-    bpf_probe_read(&mm, sizeof(mm), &t->mm);
-    bpf_probe_read(&exe_file, sizeof(exe_file), &mm->exe_file);
-    bpf_probe_read(&inode, sizeof(inode), &exe_file->f_inode);
-    if (!inode) {
-        log_debug("get_offsets_data: could not read inode struct pointer\n");
+    int err;
+    err = BPF_CORE_READ_INTO(&key.ino, t, mm, exe_file, f_inode, i_ino);
+    if (err) {
+        log_debug("get_offsets_data: could not read i_ino field\n");
         return NULL;
     }
 
-    bpf_probe_read(&sb, sizeof(sb), &inode->i_sb);
-    if (!sb) {
-        log_debug("get_offsets_data: could not read superblock struct pointer\n");
+    err = BPF_CORE_READ_INTO(&dev_id, t, mm, exe_file, f_inode, i_sb, s_dev);
```

Comment:
suggestion: do one read of the common path `BPF_CORE_READ_INTO(&key.ino, t, mm, exe_file, f_inode)` to prevent a lot of duplicate read calls.

Reply:
Good point. Done